packages/examples/main/src/react/app.tsx

```diff
@@ -92,31 +81,41 @@ export const ReactMonacoEditor: React.FC<EditorProps> = ({
 
     useEffect(() => {
         if (ref.current != null) {
-            // register Monaco languages
-            monaco.languages.register({
-                id: 'json',
-                extensions: ['.json', '.jsonc'],
-                aliases: ['JSON', 'json'],
-                mimetypes: ['application/json']
-            });
-
-            // create Monaco editor
-            editorRef.current = monaco.editor.create(ref.current!, {
-                model: monaco.editor.createModel(defaultCode, 'json', monaco.Uri.parse('inmemory://model.json')),
-                glyphMargin: true,
-                lightbulb: {
-                    enabled: true
-                },
-                automaticLayout: true
-            });
-
-            // install Monaco language client services
-            MonacoServices.install();
-
-            lspWebSocket = createWebSocket(url);
+            const createEditor = () => {
+                // register Monaco languages
+                monaco.languages.register({
+                    id: 'json',
+                    extensions: ['.json', '.jsonc'],
+                    aliases: ['JSON', 'json'],
+                    mimetypes: ['application/json']
+                });
+
+                // create Monaco editor
+                editorRef.current = monaco.editor.create(ref.current!, {
+                    model: monaco.editor.createModel(defaultCode, 'json', monaco.Uri.parse('inmemory://model.json')),
+                    glyphMargin: true,
+                    lightbulb: {
+                        enabled: true
+                    },
+                    automaticLayout: true
+                });
+
+                lspWebSocket = createWebSocket(url);
+            };
+
+            const monacoVscodeApiInit = async () => {
+                await initServices({
+                    enableThemeService: true
+                }).then(() => createEditor());
+            };
+            if (init) {
+                monacoVscodeApiInit();
+            } else {
+                createEditor();
+            }
```

Comment:
Where is init set to false?

Also the syntax pretty confusing

what about:
```typescript
const initAndCreateEditor = async () => {
  if (init) {
    await initServices({
      enableThemeService: true
    })
    init = false
  }
  createEditor();
}
```

or even better:

```typescript
const vscodeApiInitPromise = initServices({
  enableThemeService: true
})

[...]

  await vscodeApiInitPromise
  createEditor();
```

Reply:
What about this:
```typescript
if (init) {
	(async () => {
		await initServices({
			enableThemeService: true
		});
		createEditor();
		init = false;
	})();
} else {
	createEditor();
}
```

Reply:
Implemented as pointed out ⬆️, see commit below ⬇️